scripts/viewSnappBaseForCurrentIndex.js

```diff
@@ -0,0 +1,22 @@
+const SnappBase = artifacts.require("SnappBase")
+
+// This script allows to view data from the SnappBase contract
+// example for viewing the current stateRoots: truffle exec scripts/viewSnappBaseForCurrentIndex.js 'stateRoots'
+
+module.exports = async (callback) => {
+  try {
+    const arguments = await process.argv.slice(4)
+    if (arguments.length != 1) {
+      callback("Error: This script requires arguments - <targetStateHash>")
+    }
+    const [functionName] = arguments
+    
+    const instance = await SnappBase.deployed()
+    const state_index = (await instance.stateIndex.call()).toNumber()
+    const info = await instance[functionName].call(state_index)
```

Comment:
I don't like that we can pass any function but hardcode that it takes state_index as an argument. This effectively limits this script to only one function. We could consider adding a SC function getCurrentState and use that function (without any argument). Also this script could then be called invokeViewFunction

Reply:
good suggestion